app/controllers/application_controller.rb

```diff
@@ -1,10 +1,156 @@
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
-
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    use Rack::Flash
+    set :session_secret, "ek_and_mk_for_lyfe"
+  end
+
+  helpers do
+    def current_user
+      User.find(session[:user_id])
+    end
+
+    def is_logged_in?(session)
+      session.has_key?(:user_id)
+    end
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  get '/tweets' do
+    if is_logged_in?(session)
+      @tweets = Tweet.all.sort_by {|tweet| tweet.id}.reverse
+      erb :'/tweets/tweets'
+    else
+      flash[:error] = "Please log in to view tweets."
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if is_logged_in?(session)
+      erb :'/tweets/create_tweet'
+    else
+      flash[:error] = "Please log in or create an account to start tweeting."
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content] != ""
+      Tweet.create(content: params[:content], user_id: session[:user_id])
+      flash[:success] = "Tweet posted successfully!"
+      redirect '/tweets'
+    else
+      flash[:error] = "Tweets are short, but not that short. Make sure you have text in your tweet!"
+      redirect '/tweets/new'
+    end
+  end
+
+  get '/tweets/:id' do
+    if is_logged_in?(session)
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'/tweets/show_tweet'
+    else
+      flash[:error] = "Please log in to view tweets."
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    @tweet = Tweet.find_by_id(params[:id])
+    if is_logged_in?(session) && @tweet.user_id == current_user.id
+      erb :'/tweets/edit'
+    elsif is_logged_in?(session) && @tweet.user_id != current_user.id
+      flash[:error] = "Hands off that tweet! Remember, you can only edit your own tweets."
+      redirect '/tweets'
+    else
+      flash[:error] = "Please log in before editing any tweets."
+      redirect '/login'
+    end
+  end
+
+  patch '/tweets/:id' do
+    tweet = Tweet.find_by_id(params[:id])
+    if params[:content] != ""
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...